pennylane/ops/qubit/parametric_ops_single_qubit.py

```diff
@@ -113,6 +114,15 @@ def compute_matrix(theta: TensorLike) -> TensorLike:  # pylint: disable=argument
         js = -1j * s
         return qml.math.stack([stack_last([c, js]), stack_last([js, c])], axis=-2)
 
+    @staticmethod
+    def compute_sparse_matrix(theta):
+        return sp.sparse.csr_matrix(
+            [
+                [np.cos(theta / 2), -1j * np.sin(theta / 2)],
+                [-1j * np.sin(theta / 2), np.cos(theta / 2)],
+            ]
+        )
+
```

Comment:
I wonder if we have any observable improvement in performance with representing a 2x2 dense matrix as a scipy sparse matrix. Some profiling, with some typical uses, would be nice to check that.

Reply:
I don't think there will be improvement in 2by2 case. But there will be quite some things when it comes to larger composite ops like 20 different QU tensor producted together. I had a private ongoing work to test this; will summarize and post in sparse epic channel whne it's readable

Reply:
This is great, thanks!